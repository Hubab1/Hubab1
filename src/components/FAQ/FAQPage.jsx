import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'app/constants';
import { sessionIsValidForCommunityId } from 'utils/misc';
import captureRoute from 'app/captureRoute';

import UnauthenticatedPage from 'components/common/Page/UnauthenticatedPage';
import FAQ from 'components/FAQ/FAQ';

export function FAQPage({ config, isSignedIn }) {
    if (!isSignedIn) {
        return (
            <UnauthenticatedPage>
                <FAQ community={config.community} />
            </UnauthenticatedPage>
        );
    }

    return (
        <FAQ community={config.community}/>
    );
}

FAQPage.propTypes = {
    config: PropTypes.object.isRequired,
    isSignedIn: PropTypes.bool.isRequired,
}

const mapStateToProps = (state) => ({
    config: state.configuration,
    isSignedIn: sessionIsValidForCommunityId(state.siteConfig.basename),
});

export default connect(mapStateToProps, null)(captureRoute(FAQPage, ROUTES.FAQ));
