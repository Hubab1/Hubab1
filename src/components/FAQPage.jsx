import React from 'react';

import { ScrollableTermsCardSection, Card, H1 } from 'assets/styles';

export function FAQPage() {
    return (
        <>
            <H1>Frequently Asked Questions</H1>
            <br/>
            <Card>
                <ScrollableTermsCardSection>
                </ScrollableTermsCardSection>
            </Card>
        </>
    )
}

export default FAQPage
