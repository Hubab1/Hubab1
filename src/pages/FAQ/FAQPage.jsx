import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'constants/constants';
import { sessionIsValidForCommunityId } from 'utils/misc';
import captureRoute from 'utils/captureRoute';

import UnauthenticatedPage from 'pages/Unauthenticated';
import FAQ from 'pages/FAQ/components/FAQ';

export function FAQPage({ config, isSignedIn }) {
    if (!isSignedIn) {
        return (
            <UnauthenticatedPage>
                <FAQ community={config.community} />
            </UnauthenticatedPage>
        );
    }

    return <FAQ community={config.community} />;
}

FAQPage.propTypes = {
    config: PropTypes.object.isRequired,
    isSignedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
    config: state.configuration,
    isSignedIn: sessionIsValidForCommunityId(state.siteConfig.basename),
});

export default connect(mapStateToProps, null)(captureRoute(FAQPage, ROUTES.FAQ));
