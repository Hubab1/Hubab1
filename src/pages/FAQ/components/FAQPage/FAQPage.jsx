import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'constants/constants';
import { sessionIsValidForCommunityId } from 'utils/misc';
import captureRoute from 'utils/captureRoute';

import UnauthenticatedPage from 'components//Pages/UnauthenticatedPage/UnauthenticatedPage';
import FAQ from '../FAQ/FAQ';

export function FAQPage({ config, isSignedIn }) {
    if (!isSignedIn) {
        return (
            <UnauthenticatedPage>
                <FAQ community={config.community} />
            </UnauthenticatedPage>
        );
    }

    return <FAQ community={config.community} />;
}

FAQPage.propTypes = {
    config: PropTypes.object.isRequired,
    isSignedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
    config: state.configuration,
    isSignedIn: sessionIsValidForCommunityId(state.siteConfig.basename),
});

export default connect(mapStateToProps, null)(captureRoute(FAQPage, ROUTES.FAQ));
