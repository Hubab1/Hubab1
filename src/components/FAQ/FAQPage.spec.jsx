import React from 'react';
import { shallow } from 'enzyme';
import { FAQPage } from './FAQPage';

describe('FAQPage', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(
            <FAQPage isSignedIn={true} config={{ community: { building_name: 'Community Name' } }} />
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
