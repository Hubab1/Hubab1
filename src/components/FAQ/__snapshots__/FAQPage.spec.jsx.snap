// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQPage matches snapshot 1`] = `
<FAQ
  community={
    Object {
      "building_name": "Community Name",
    }
  }
/>
`;
